### ISSUE TITLE
3.9.3 makeoftexe_test and makeotf_test failures.

### ISSUE THREAD

**Issue by NSGod** [adobe-type-tools/afdko#1621]
I've got several bug fixes I'd like to contribute but I can't get the 3.9.3 tests to complete without error even without my fixes. I'm on macOS Ventura 13.2.1 with Xcode 14.3 beta. I have Python 3.11.2 installed through MacPorts.

One specific reproducible problem is the following using a fresh afdko 3.9.3 setup (alongside it, I have a checked out clone of 3.9.3 of the project). Here, I'm simply picking out a specific command from the makeotfexe tests that causes a crash:

```
mdouma46% python -m venv afdko_ENV; cd afdko_ENV; source bin/activate
(afdko_ENV) mdouma46% pip install afdko
(afdko_ENV) mdouma46% makeotfexe -f ../afdko/tests/makeotfexe_data/input/bug965/font.pfa -ff ../afdko/tests/makeotfexe_data/input/bug965/feat.fea -o "/tmp/tmpOutput.otf"

```

`makeotfexe` crashes with the following stack trace:

```
Thread 0 Crashed::  Dispatch queue: com.apple.main-thread
0   makeotfexe                    	       0x104373e89 dnaNext + 9
1   makeotfexe                    	       0x1043ec4af otlSubtableAdd + 47
2   makeotfexe                    	       0x1043ce199 GPOSFill + 361
3   makeotfexe                    	       0x1043ecb7a sfntFill + 122
4   makeotfexe                    	       0x1043df78b hotConvert + 2859
5   makeotfexe                    	       0x10436fc43 cbConvert + 1811
6   makeotfexe                    	       0x10437125b parseArgs + 3707
7   makeotfexe                    	       0x104370359 c_main + 377
8   makeotfexe                    	       0x10436de89 main + 9
9   dyld                          	    0x7ff80d3e0310 start + 2432

```

Basically, the `GPOSCtx.otl table` is `NULL` and calling `dnaNEXT(t->subtable)` dereferences a NULL pointer. I'm wondering if anyone else can reproduce this crash?

There are a couple of other issues (`No such file or directory: 'makeotf'`) that I'm not sure if I'm not running the tests properly. 

Not having contributed to this project before, I guess I'm not sure how seriously to take the test failures. Like if I can't get the code to pass tests locally, should I hold off on submitting my bug fixes, or?

Thanks

Mark Douma

**Comment by skef** [comment]
First thing to do (unless you've already done it and are trying to contribute Python 3.11 fixes) is to try everything against Python 3.10 or a bit earlier. There are known problems with 3.11 and we're waiting to let the tool-chain and dependencies bake a while before we try to get everything working with it.

**Comment by NSGod** [comment]
I've managed to get the tests working w/ Python 3.10 and using:

```
pip install wheel
pip install .
```
I then ran the tests using Visual Studio Code's UI. All tests complete fine except for `test_infinite_loop_with_dflt_lookups_bug965` which crashes.

Previously, I had tried generating an Xcode project and running the tests from the test target which didn't seem to work too well.
